regnety/config/config.py

```diff
@@ -35,12 +38,105 @@ class RegNetYConfig:
     wm: float
 
 
+@dataclass
+class TrainConfig:
+    """
+    Dataclass of training configuration for RegNetY
+
+    Args:
+        optimizer: One of "sgd", "adam", "adamw"
+        base_lr: Base learning rate for training
+        warmup_epochs: Number of epochs used for warmup
+        warmup_factor: Gradual linear warmup factor
+        total_epochs: Number of training epochs
+        weight_decay: Weight decay to be used in optimizer
+        momentum: Momentum to be used in optimizer
+        lr_schedule: One of "constant" or "half_cos"
+        log_dir: Path to store logs
+        model_dir: Path to store model checkpoints
+    """
+
+    optimizer: str
+    base_lr: float
+    warmup_epochs: int
+    warmup_factor: float
+    total_epochs: int
+    weight_decay: float
+    momentum: float
+    lr_schedule: str
+    log_dir: str
+    model_dir: str
+    cache_dir: str
+
+
+@dataclass
+class PreprocessingConfig:
+    tfrecs_filepath: List[str]
+    batch_size: int
+    image_size: int
+    crop_size: int
+    resize_to_size: int
+    augment_fn: Union[str, Callable]
+    num_classes: int
+    cache_dir: str
+    color_jitter: bool
+    scale_to_unit: bool
+    scale_method: str
+
+
+def get_preprocessing_config(
+    tfrecs_filepath: List[str] = None,
+    batch_size: int = 1024,
+    image_size: int = 512,
+    crop_size: int = 224,
+    resize_to_size: int = 320,
+    augment_fn: Union[str, Callable] = "default",
+    num_classes: int = 1000,
+    cache_dir: str = "gs://adityakane-train/cache/",
+    color_jitter: bool = False,
+    scale_to_unit: bool = True,
+    scale_method: str = "tf"
+):
+
+    """
+    Getter function for preprocessing configuration.
+
+    Args:
+        tfrecs_filepath: list of filepaths of all TFRecords files.
+        batch_size: batch_size for the Dataset.
+        image_size: final image size of the images in the dataset.
+        augment_fn: function to apply to dataset after loading raw TFrecords.
+        num_classes: number of classes.
+        percent_valid: Percentage of training data to be used as validation data.
+        cache_dir: Directory to use for TF cache. 
+        color_jitter: If True, color_jitter augmentation is applied.
+        scale_to_unit: Whether the images should be scaled to [0,1].
+    
+    Returns:
+        A PreprocessingConfig dataclass instance with all attributes
+    
+    """
+    return PreprocessingConfig(
+        tfrecs_filepath=tfrecs_filepath,
+        batch_size=batch_size,
+        image_size=image_size,
+        crop_size=crop_size,
+        resize_to_size=resize_to_size,
```

Comment:
Please clarify in the Pydoc why both crop_size and resize_to_size are needed, and what the difference is.

Reply:
Thanks for explaining this. I think `resize_pre_crop` would be a better name in that case.